wagtail/images/models.py

```diff
@@ -509,14 +499,220 @@ def create_rendition(self, filter: "Filter") -> "AbstractRendition":
         )
         return rendition
 
-    def generate_rendition_file(self, filter: "Filter") -> File:
+    def get_renditions(self, *filter_specs: str) -> Dict[str, "AbstractRendition"]:
+        """
+        Returns a ``dict`` of ``Rendition`` instances with image files reflecting
+        the supplied ``filter_specs``, keyed by the relevant ``filter_spec`` string.
+
+        Note: If using custom image models, instances of the custom rendition
+        model will be returned.
+        """
+        Rendition = self.get_rendition_model()
+        filters = tuple(Filter(spec) for spec in set(filter_specs))
```

Comment:
```suggestion
        filters = [Filter(spec) for spec in set(filter_specs)]
```

Reply:
note that `set(filter_specs)` will change the order of the passed filter_specs. for example

```pycon
>>> set(['fill-180x180-c100', 'max-300x300', 'width-600'])
{'max-300x300', 'width-600', 'fill-180x180-c100'}
```

I would say we should preserve the order otherwise it will create a great deal of confusion

Reply:
`filters = [Filter(spec) for spec in dict.fromkeys(filter_specs).keys()]` should do it